src/llmcompressor/entrypoints/oneshot.py

```diff
@@ -174,8 +174,160 @@ def apply_recipe_modifiers(
         session.finalize(**session_kwargs)
 
 
-def oneshot(**kwargs) -> PreTrainedModel:
-    one_shot = Oneshot(**kwargs)
+def oneshot(
+    # Required model parameters
+    model: str,
```

Comment:
Users can also pass in a pre-loaded model
```suggestion
    model: Union[str, PreTrainedModel],
```